test/extensions/network/dns_resolver/cares/dns_impl_test.cc

```diff
@@ -396,6 +396,41 @@ TEST_F(DnsImplConstructor, SupportsCustomResolvers) {
   ares_free_data(resolvers);
 }
 
+TEST_F(DnsImplConstructor, SupportsCustomResolversAsFallback) {
+  char addr4str[INET_ADDRSTRLEN];
+  auto addr4 = Network::Utility::parseInternetAddress("1.2.3.4");
+
+  // convert the address and options into typed_dns_resolver_config
+  envoy::config::core::v3::Address dns_resolvers;
+  Network::Utility::addressToProtobufAddress(
+      Network::Address::Ipv4Instance(addr4->ip()->addressAsString(), addr4->ip()->port()),
+      dns_resolvers);
+  envoy::extensions::network::dns_resolver::cares::v3::CaresDnsResolverConfig cares;
+  cares.set_use_resolvers_as_fallback(true);
+  cares.add_resolvers()->MergeFrom(dns_resolvers);
+
+  // copy over dns_resolver_options_
+  cares.mutable_dns_resolver_options()->MergeFrom(dns_resolver_options_);
+
+  envoy::config::core::v3::TypedExtensionConfig typed_dns_resolver_config;
+  typed_dns_resolver_config.mutable_typed_config()->PackFrom(cares);
+  typed_dns_resolver_config.set_name(std::string(Network::CaresDnsResolver));
+  Network::DnsResolverFactory& dns_resolver_factory =
+      createDnsResolverFactoryFromTypedConfig(typed_dns_resolver_config);
+  auto resolver =
+      dns_resolver_factory.createDnsResolver(*dispatcher_, *api_, typed_dns_resolver_config);
+
+  // Given that the local machine will have a working conf in resolve.conf the resolver will not
+  // use the fallback given.
```

Comment:
Any way to test the inverse? A case where we'll use the fallback

Reply:
I tried for a bit, but couldn't defeat all the fallbacks available on linux, i.e., I was not able to break the machine enough to get to init_by_defaults.

FWIW I have tested on android where the fallback _is_ used. I can try to see if I can get there with a unit test again.